src/cron/hourlyNotifications.ts

```diff
@@ -0,0 +1,72 @@
+import { utcToZonedTime } from 'date-fns-tz';
+import {
+  schedulePersonalizedDigestSubscriptions,
+  notifyGeneratePersonalizedDigest,
+} from '../common';
+import {
+  User,
+  UserPersonalizedDigest,
+  UserPersonalizedDigestSendType,
+  UserPersonalizedDigestType,
+} from '../entity';
+import { Cron } from './cron';
+import { isWeekend, subDays } from 'date-fns';
+import { DEFAULT_TIMEZONE } from '../types';
+
+const sendType = UserPersonalizedDigestSendType.workdays;
+const digestTypes = [UserPersonalizedDigestType.StreakReminder];
+
+const atTimeZone =
+  "AT TIME ZONE COALESCE(NULLIF(u.timezone, ''), :defaultTimezone)";
+
+const cron: Cron = {
+  name: 'hourly-notification',
+  handler: async (con, logger) => {
+    const personalizedDigestQuery = con
+      .createQueryBuilder()
+      .select('upd.*, u.timezone')
+      .from(UserPersonalizedDigest, 'upd')
+      .innerJoin(User, 'u', 'u.id = upd."userId"')
+      .where(
+        `clamp_to_hours(upd."preferredHour" - EXTRACT(HOUR FROM NOW() ${atTimeZone})) = 1`,
+        {
+          defaultTimezone: DEFAULT_TIMEZONE,
+        },
+      )
+      .andWhere(`flags->>'sendType' = :sendType`, { sendType })
+      .andWhere(`upd.type in (:...digestTypes)`, { digestTypes });
+
+    // Make sure digest is sent at the beginning of the hour
+    const timestamp = new Date().getTime();
```

Comment:
You need to move this inside `handler` right to make the time be exact at when the cron schedules?

Reply:
Like this the timestamp will still be XX:58

Reply:
I just used the one inside the worker that the cron triggers
https://github.com/dailydotdev/daily-api/blob/ce43ed767acbc6dfd260fc36e5c5298adb9f65ab/src/workers/personalizedDigestEmail.ts#L155
https://github.com/dailydotdev/daily-api/blob/ce43ed767acbc6dfd260fc36e5c5298adb9f65ab/src/workers/personalizedDigestEmail.ts#L185

Reply:
I moved it into the handler, though it doesn't use it for anything except checking if it is weekend. The `previousSendTimestamp`, `emailSendTimestamp` and `emailBatchId` have no effect on the worker

Reply:
It would be good to have it standardized from outside, at least `emailSendTimestamp` so it controls when worker sends for hourly. I get why you don't need `previousSendTimestamp`.

Reply:
That way the worker is not coupled with specific send schedule, it just takes the `emailSendTimestamp` and sends.

Reply:
Could also do as @rebelchris suggested, omit `send_after` in the onesignal function. So we don't need the timestamp for this worker?

Reply:
I removed the send_after config for streak reminder function, so it gets sent immediately after invocation.
Does that work?

Reply:
Just omit it yeah.
It will send directly which I think is fine right?

Reply:
👍